scripts.Rmd

```diff
@@ -0,0 +1,255 @@
+tmp <- file.copy("script-based-workflows/raw_data.xlsx", ".")
+tmp <- file.copy("script-based-workflows/report.Rmd", ".")
+dir.create("data")
+```
+
+`drake` has been built with the firm belief that the best workflows rely on approaching problems from a function perspective. This idea is supported in "Good enough practices in scientific computing", that approaching your project by writing modular code increases the end product.
+
+> The core realization in these practices is that being readable, reusable, and testable are all side effects of writing modular code, i.e., of building programs out of short, single-purpose functions with clearly-defined inputs and outputs 
+> 2017 Wilson et al.
+
+That being said, not all projects and workflows are written using this mantra and for others the lift would be too great to re-factor to a purely functional approach. 
+
+## Traditional and Legacy Workflows
+
+A common approach to creating workflows is to write steps of the analysis into scripts that perform specific tasks. By giving these scripts meaningful names, e.g. 01_data.R, 02_munge.R, 03_analysis.R,  one can surmise both their contents and the order in which these scripts should be run.
+
+It was acknowledged in the (What we left out)[https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec027] section that build tools to manage projects when code gets out of date are ideal, but can be substituted with a simple master script called `make.R` that reruns each of the scripts in order.
+
+```{r,eval=FALSE}
+
+# Preprocessing steps
+source("01_data.R")
+source("02_munge.R")
+source("03_histogram.R")
+source("04_regression.R")
+
+# Calls rmarkdown::render on report.Rmd
+source("05_report.R")
+```
+
+The argument being that while some code may be reran, for small projects this likely is not going to be a bottleneck. 
+
+## Overcoming Technical Debt
+
+The argument against the workflow manager as above is the whole reason `drake` was developed: For non-trivial or long running processes, simply rerunning all the steps is unfeasible.
+
+> With the `code_to_function()` we can now move these encumbered workflows into the managed plan ecosystem of `drake`.
+
+`code_to_function()` accepts the path to either an R or Rmd script and will generate a function that contains the contents of the script you provided as a function.
+
+
+```{r}
+
+load_data <- code_to_function("01_data.R")
+
+```
+
+By turning these scripts into functions, they can be added into a `drake` plan for workflow management.
+
+
+## Dependencies
+
+An important aspect of drake plans is the capability to track the dependencies of each function and only rerun the parts of the plan that have updated, either in code or input. Dependencies in the terms of `drake` means to rely on prior steps to have been run and generated output that will then be used in the current step. Any time the prior step updates, then any down-stream steps that rely on its outputs need to be updated.
```

Comment:
Suggestion: "Dependencies in the terms of `drake` means to rely on prior steps to have been run and generated output that will then be used in the current step." => "In `drake`, a target's dependencies are the things it needs in order to build. Dependencies can include functions, files, and other targets upstream."